mlflow/server/js/src/components/MetricsPlotView.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import {
+  Bar,
+  BarChart,
+  CartesianGrid,
+  Legend,
+  Line,
+  LineChart,
+  Tooltip,
+  XAxis,
+  YAxis,
+  ResponsiveContainer,
+} from 'recharts';
+import Utils from '../utils/Utils';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { X_AXIS_STEP, X_AXIS_RELATIVE, X_AXIS_WALL } from './MetricsPlotControls';
+import { CHART_TYPE_BAR } from './MetricsPlotPanel';
+
+const COLORS = [
+  '#A3C3D9',
+  '#82ca9d',
+  '#8884d8',
+  '#FF82A9',
+  '#FFC0BE',
+  '#AE76A6',
+  '#993955',
+  '#364958',
+];
+
+export class MetricsPlotView extends React.Component {
+  static propTypes = {
+    runUuids: PropTypes.arrayOf(String).isRequired,
+    metrics: PropTypes.arrayOf(Object).isRequired,
+    xAxis: PropTypes.string.isRequired,
+    metricKeys: PropTypes.arrayOf(String).isRequired,
+    showDot: PropTypes.bool.isRequired,
+    chartType: PropTypes.string.isRequired,
+  };
+
+  static MAX_RUN_NAME_DISPLAY_LENGTH = 36;
+
+  getIdForRunMetricData = (runUuid, metricKey) => `${runUuid}_${metricKey}`;
+
+  parseTimestamp = (timestampStr, baseTimestamp, xAxis) => {
+    const timestamp = Number.parseFloat(timestampStr);
+    return xAxis === X_AXIS_RELATIVE ? (timestamp - baseTimestamp) / 1000 : timestamp;
```

Comment:
BTW, I'm guessing you probably already discussed this with others - do we intend for this behavior? It seems to show each metric relative to its run start during run comparison, which I guess actually makes sense, but I was confused about what exactly was happening in the default view (relative timestamp as X axis) initially. Maybe we could/should add a tooltip explaining what relative time means? Could defer it to a follow-up PR too.

Also, would it make sense to have a unit for the relative timestamp x axis (so that it's clear it's in ms)?